airbyte-server/src/main/java/io/airbyte/server/config/ApplicationBeanFactory.java

```diff
@@ -109,6 +110,11 @@ public AirbyteProtocolVersionRange airbyteProtocolVersionRange(
     return new AirbyteProtocolVersionRange(new Version(minVersion), new Version(maxVersion));
   }
 
+  @Singleton
```

Comment:
Why is it needed in the server? It feel strange to have a need for a client in the server. Is the server doing request to the server?

Reply:
It is used by the `OAuthHandler`, presumably to fetch OAuth tokens.